script.js

```diff
@@ -47,4 +50,119 @@ buttonElement.addEventListener('click', function (event) {
     subElement.classList.toggle('active');
 });
 
+buttonRight.addEventListener('click', function (){
+    move('right');
+}); 
+buttonLeft.addEventListener('click', function (){
+    move('left');
+});
+
+let images = document.querySelectorAll('.carousel object');
+let slider = [];
+for (let i = 0; i < images.length; i++) {
+    slider[i] = images[i].data;
+    images[i].remove();
+}
+let boxes = document.querySelectorAll('.carousel__box');
+for (let i = 0; i < boxes.length; i++) {
+    boxes[i].remove();
+}
+
+let carouselTextObject = document.createElement('div');
+carouselTextObject = carouselText;
+
+let step = 0;
+let carousel =  Array.from(document.querySelectorAll('.carousel__box'));
 
+function carouselObject (position, left) {
+    let objectDiv = document.createElement('div');
+    let objectImg = document.createElement('object');
+    objectImg.data = slider[position];
+    objectImg.classList.add('carousel__object');
+    objectDiv.classList.add('carousel__box');
+    objectDiv.style.left = left;
+    if (position === 0) {
+        objectDiv.appendChild(carouselTextObject);
+    }
+    objectDiv.appendChild(objectImg);
+    document.querySelector('.carousel').appendChild(objectDiv);
+    carousel.push(objectDiv);
+}
+
+function draw() {
+    carouselObject (0, "0%");
+    carouselObject (1, "100%");
+    carouselObject (2, "-100%");
+};
+draw();
+
+function drawRight() {
+    let position = step - 1;
+    if (position < 0) {
+        position = slider.length - 1;
+    }
+    carouselObject (position, "100%");
+    step++;
+    if (step === slider.length) {
+        step = 0;
+    }
+};
+
+function drawLeft() {
+    let position = step + 1;
+    if (position === slider.length) {
+        position = 0;
+    }
+    carouselObject(position, "-100%");
+    step--;
+    if (step < 0) {
+        step = slider.length - 1;
+    }
+};
+
+function move(direction) {
+    if (direction === 'right') {
+        buttonRight.disabled = true;
+        buttonLeft.disabled = true;
+        for (let i = 0; i < carousel.length; i++) {
+            if (carousel[i].style.left === "-100%") {
+                carousel[i].remove();
+            }
+        }
+        for (let i = 0; i < carousel.length; i++) {
+            if (carousel[i].style.left === "0%") {
+                carousel[i].style.left = "-100%";
+            }
+            else {
+                carousel[i].style.left = "0%";
+            }
+        }
+        setTimeout(function() {
+            drawRight(); 
+            buttonRight.disabled = false;
+            buttonLeft.disabled = false;
+        }, 500);
```

Comment:
This blocks carousel from being used while  transition is happening, While not ideal solution this will work, you can leave it as is, or if you want try to figure out how to queue those requests for multiple button slicks.

Reply:
Ok, I will study asynch methods. I think I can change it.